### PR TITLE
Shelve files from preservation

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?
ref https://github.com/sul-dlss/common-accessioning/issues/754

### CODE REVIEW

File: spec/services/shelvable_files_stager_spec.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  let(:workspace_druid) { DruidTools::Druid.new(druid, workspace_root) }
+
+  # create an empty workspace location for object content files
+  let(:content_dir) { Pathname(workspace_druid.path('content', true)) }
+
+  # read in a FileInventoryDifference manifest from the fixtures area
+  let(:content_diff_reports) { Pathname('spec').join('fixtures', 'content_diff_reports') }
+  let(:inventory_diff_xml) { content_diff_reports.join('ng782rw8378-v3-v4.xml') }
+  let(:inventory_diff) { Moab::FileInventoryDifference.parse(inventory_diff_xml.read) }
+  let(:content_diff) { inventory_diff.group_difference('content') }
```
Comment (jmartin-sul) [inline spec/services/shelvable_files_stager_spec.rb:20]: similar question to the one for the class being tested, should this actually be named `shelve_diff`?

**Review by jmartin-sul — COMMENTED** [review]
some nitpicky questions about naming, declaration, and test detritus cleanup, but overall LGTM

File: spec/services/shelvable_files_stager_spec.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ShelvableFilesStager do
+  subject(:stage) { described_class.stage(druid, content_metadata, content_diff, content_dir) }
+
+  let(:workspace_root) { Dir.mktmpdir }
```
[the remark this answers is not in the document]
Comment (jcoyne) [reply spec/services/shelvable_files_stager_spec.rb:8]: Yes, tmp gets cleaned up automatically on most systems.  This is a copy from the existing test from `spec/services/shelving_service_spec.rb`

**Comment by jcoyne** [comment]
@jmartin-sul Thank you for your thoughtful review. I've made the changes you recommended.

**Review by jmartin-sul — APPROVED** [review]
thanks for the changes!

**Pull request merged**